aips/aip-order-book.md

```diff
@@ -0,0 +1,326 @@
+   GTC orders are orders that are valid until they are cancelled or filled. Post Only orders are orders that are valid only if they are not
+   taker orders. IOC orders are orders that are valid only if they are taker orders.
+
+   In addition, the market also supports trigger conditions for orders. An order with trigger condition is not put
+   on the order book until its trigger conditions are met. Following trigger conditions are supported:
+   - TakeProfit(price): If its a buy order its triggered when the market price is greater than or equal to the price. If
+      its a sell order its triggered when the market price is less than or equal to the price.
+   - StopLoss(price): If its a buy order its triggered when the market price is less than or equal to the price. If its
+      a sell order its triggered when the market price is greater than or equal to the price.
+   - TimeBased(time): The order is triggered when the current time is greater than or equal to the time.
+3. Pluggable Clearinghouse - The clearinghouse is responsible for managing the settlement of trades and actual transfer of assets between trading parties. The clearinghouse is pluggable,
+   allowing for different implementations to be used depending on the specific needs of the exchange (Spot vs Perps for example).
+   A clearing house implementation is expected to implement the following APIs
+      - settle_trade(taker, maker, is_taker_long, price, size): SettleTradeResult -> Called by the market when there
+      is an match between taker and maker. The clearinghouse is expected to settle the trade and return the result. Please 
+      note that the clearing house settlment size might not be the same as the order match size and the settlement might also fail.
+      - validate_settlement_update(account, is_taker, is_long, price, size): bool -> Called by the market to validate
+      an order when its placed. The clearinghouse is expected to validate the order and return true if the order is valid.
+      - max_settlement_size(account, is_long, orig_size): Option<u64> -> Called by the market to validate the size
+      of the order when its placed. The clearinghouse is expected to validate the order and return the maximum settlement size
+      of the order. Checkout clearinghouse_test as an example of the simplest form of clearing house implementation that just tracks
+      the position size of the user and does not do any validation.
+## Impact
+
+- Anyone who wants to build a high-throughput DEX
+
+## Alternative Solutions
+
+Keeping GTE on the ecosystem to implement, might not allow for all the optimizations that can be done with tight integration with the VM/BlockSTM.
+
+## Specification and Implementation Details
+
+Following are the main APIs supported by the CLOB
+
+```
+   struct OrderBook<M: store + copy + drop> has store;
+
+    struct SingleOrderMatch<M: store + copy + drop> has drop, copy {
+        order: Order<M>,
+        matched_size: u64
+    }
```

Comment:
curious why we didn't use objects

Reply:
Can you elaborate.. how do you envision we using objects here?